### PR TITLE
added open dyslexic font

### PR DESCRIPTION
**Pull request by 27vinayvyas**
This PR is regarding #1606 and does the following changes
1 Create new dropdown item in TopToolbar.java with the corresponding changes in the OdeMessages.java file
2 Add OpenDyslexic font files in appengine/war/fonts
3 Required css changes in Ya.css and gwt.css files (Changes discussed in 'issues')

### CODE REVIEW

**Comment by AppInventorWorkerBee** [comment]
Can one of the admins verify this patch?

**Comment by 27vinayvyas** [comment]
I deleted google-cloud-sdk in this second commit

File: appinventor/appengine/src/com/google/appinventor/client/OdeMessages.java [hunk 1]
```diff
@@ -664,6 +664,37 @@
   @Description("Link for Report an Issue form")
   String feedbackMenuItem();
 
+
```
Comment (ewpatton) [inline appinventor/appengine/src/com/google/appinventor/client/OdeMessages.java]: Reduce the amount of whitespace added here.

File: appinventor/appengine/src/com/google/appinventor/client/TopToolbar.java [hunk 2]
```diff
@@ -6,6 +6,7 @@
 
 package com.google.appinventor.client;
 
+import com.google.gwt.user.client.ui.RootPanel;
```
Comment (ewpatton) [inline appinventor/appengine/src/com/google/appinventor/client/TopToolbar.java]: Keep the imports alphabetized. This should be closer to line 57.

File: appinventor/appengine/src/com/google/appinventor/client/TopToolbar.java [hunk 3]
```diff
@@ -144,9 +151,9 @@
   public TopToolbar() {
     /*
      * Layout is as follows:
-     * +--------------------------------------------------+
-     * | Project ▾ | Connect ▾ | Build ▾| Help ▾| Admin ▾ |
-     * +--------------------------------------------------+
+     * +---------------------------------------------------------------+
+     * | Project ▾ | Connect ▾ | Build ▾| Help ▾| Admin ▾ | Dyslexic ▾ |
```
Comment (ewpatton) [inline appinventor/appengine/src/com/google/appinventor/client/TopToolbar.java]: I wonder if maybe it is better to have this called "Settings" as I expect that we will want to have more in the future, and I don't know that we want to consume more primary menu space for an option that only affects a small portion of app inventors.

File: appinventor/appengine/src/com/google/appinventor/client/TopToolbar.java [hunk 4]
```diff
@@ -307,21 +325,28 @@ public TopToolbar() {
...
     toolbar.add(fileDropDown);
     toolbar.add(connectDropDown);
     toolbar.add(buildDropDown);
+    
+    toolbar.add(dyslexicDropDown); 
 
     // Commented out language switching until we have a clean Chinese translation. (AFM)
     toolbar.add(helpDropDown);
 
+
```
Comment (ewpatton) [inline appinventor/appengine/src/com/google/appinventor/client/TopToolbar.java]: Remove extraneous whitespace.

File: appinventor/appengine/src/com/google/appinventor/client/TopToolbar.java [hunk 5]
```diff
@@ -345,6 +370,57 @@ public TopToolbar() {
...
+
+
+
+ private static class changeFontDys implements Command {
+    @Override
+    public void execute() {
+      RootPanel.get().addStyleName("gwt-bodyOdf");
+      RootPanel.get().removeStyleName("gwt-bodyRob");
+    // RootPanel.get().getStyle().setBackgroundColor("black");
```
Comment (ewpatton) [inline appinventor/appengine/src/com/google/appinventor/client/TopToolbar.java]: Remove commented out code.

File: appinventor/appengine/src/com/google/appinventor/client/TopToolbar.java [hunk 6]
```diff
@@ -345,6 +370,57 @@ public TopToolbar() {
   // List of Commands for use in Drop-Down Menus
   // -----------------------------
 
+
```
Comment (ewpatton) [inline appinventor/appengine/src/com/google/appinventor/client/TopToolbar.java]: Remove extra whitespace.

File: appinventor/appengine/src/com/google/appinventor/client/TopToolbar.java [hunk 7]
```diff
@@ -345,6 +370,57 @@ public TopToolbar() {
...
+      RootPanel.get().removeStyleName("gwt-bodyRob");
+    // RootPanel.get().getStyle().setBackgroundColor("black"); 
+    // RootPanel.get().removeStyleName("body-Roboto");
+     //RootPanel.get().addStyleName(".gwt-body");
+      
+     
+    }
+  }
+
+private static class changeFontRob implements Command {
```
Comment (ewpatton) [inline appinventor/appengine/src/com/google/appinventor/client/TopToolbar.java]: Indentation.

File: appinventor/appengine/src/com/google/appinventor/client/TopToolbar.java [hunk 8]
```diff
@@ -345,6 +370,57 @@ public TopToolbar() {
...
+
+
+
+
+
+
+
+
+
+ private static class changeFontDys implements Command {
```
Comment (ewpatton) [inline appinventor/appengine/src/com/google/appinventor/client/TopToolbar.java]: Indentation.

File: appinventor/appengine/src/com/google/appinventor/client/TopToolbar.java [hunk 9]
```diff
@@ -1098,3 +1174,4 @@ public void execute() {
   }
 
 }
+
```
Comment (ewpatton) [inline appinventor/appengine/src/com/google/appinventor/client/TopToolbar.java]: You don't need the extra line here, but make sure that there's a newline at the end of the file if you remove it.

File: appinventor/appengine/src/com/google/appinventor/client/opendyslexic/README.txt [hunk 10]
```diff
@@ -0,0 +1,70 @@
+README
```
Comment (ewpatton) [inline appinventor/appengine/src/com/google/appinventor/client/opendyslexic/README.txt]: The fonts are duplicated here and in appengine/war/fonts. I don't think that these need to be here and that they should be in appengine/war/fonts.

File: appinventor/appengine/war/Ya.css [hunk 11]
```diff
@@ -14,30 +14,61 @@ they get cut off in Windows.
...
+  src: url('/fonts/OpenDyslexic-Bold.otf');
 }
 
 body, html {
   height: 100%;
 }
 
+
+
+.gwt-bodyOdf{
```
Comment (ewpatton) [inline appinventor/appengine/war/Ya.css]: Put a space before the `{` on this line and the following two definitions.

File: appinventor/appengine/war/Ya.css [hunk 12]
```diff
@@ -14,30 +14,61 @@ they get cut off in Windows.
...
+  font-family: OpenDyslexic, Helvetica, Arial, sans-serif;
+}
+
+.gwt-bodyRob{
+  font-family: Roboto, Helvetica, Arial, sans-serif;
+}
+
+
+.gwt-bodyRob .gwt-bodyOdf{
```
Comment (ewpatton) [inline appinventor/appengine/war/Ya.css]: If you swap the order of `.gwt-bodyRob` and `.gwt-bodyOdf` then you may not need this rule.

File: appinventor/appengine/war/Ya.css [hunk 13]
```diff
@@ -14,30 +14,61 @@ they get cut off in Windows.
...
-  font-family: Roboto, Helvetica, Arial, sans-serif;
+  
 }
 
-table td,
-select {
-  font-family: Roboto, Helvetica, Arial, sans-serif;
-}
+
+
```
Comment (ewpatton) [inline appinventor/appengine/war/Ya.css]: Remove excess whitespace.

File: appinventor/appengine/war/Ya.css [hunk 14]
```diff
@@ -344,7 +375,7 @@ select {
 .ode-TopPanelLinks .gwt-TitleLabel {
   cursor: pointer;
   color: rgb(100, 100, 100);
-  font-family: Roboto, Helvetica, Arial, sans-serif;
+
```
Comment (ewpatton) [inline appinventor/appengine/war/Ya.css]: Remove the blank lines here and in the following CSS rules.

File: appinventor/appengine/war/Ya.css [hunk 15]
```diff
@@ -2544,4 +2573,4 @@ div.vector-marker>svg, div.leaflet-marker-icon>img {
   height: 100%;
   box-sizing: border-box;
   padding: 2px;
-}
+}
```
Comment (ewpatton) [inline appinventor/appengine/war/Ya.css]: Make sure that the file has a trailing newline.

File: appinventor/appengine/war/gwt.css [hunk 16]
```diff
@@ -13,11 +13,11 @@
  */
 
 body, table td, select {
-  font-family: Arial Unicode MS, Arial, sans-serif;
+
```
Comment (ewpatton) [inline appinventor/appengine/war/gwt.css]: Just remove the lines, don't leave the line blank.

File: appinventor/components/src/com/google/appinventor/components/runtime/AltCamera.java [hunk 17]
```diff
@@ -0,0 +1,239 @@
+// -*- mode: java; c-basic-offset: 2; -*-
```
Comment (ewpatton) [inline appinventor/components/src/com/google/appinventor/components/runtime/AltCamera.java]: This was from your part 1 task for GSOC. It shouldn't be checked in to this change.

Comment (27vinayvyas) [reply appinventor/components/src/com/google/appinventor/components/runtime/AltCamera.java]: How to do this ??
I have deleted the AltCamera File but its still available in the extension tab

**Comment by 27vinayvyas** [comment]
Please review the changes

**Comment by kangris-sket** [comment]
**I have test this pull and work fine!**
Ok


**But i get some log error/warning.. ?**
Error

**Comment by 27vinayvyas** [comment]
I would look into the matter once and get back to you in a day
Thanx for reviewing

**Comment by 27vinayvyas** [comment]
Hello 
There was an error in the file name. I have fixed it. Please review it again.
Also i might have made changes in different files in the last few days(like those of extra white spaces and similar things). Please inform me if you find those.
Thnx for reviewing

**Comment by 27vinayvyas** [comment]
Hello @kangris-sket  @ewpatton 
Please review this  PR. If any change is not required, i could move to the other issues
Thnx.

**Comment by ewpatton** [comment]
@AppInventorWorkerBee ok to test

**Comment by ewpatton** [comment]
@27vinayvyas Where did all of the .class files in lib/android/ come from? There shouldn't have been any change to that directory. Please use git checkout to revert that directory to master and then push an updated commit.

**Comment by 27vinayvyas** [comment]
@ewpatton 
I have made the required changes.
Can you please review it again

**Comment by 27vinayvyas** [comment]
Also, how to check what changes i are there in this commit. 
In the commit name(changes made) its showing 3000 changes, most of them are deleted stuff. 
From where did  you find that the unnecessary files were added in lib /android folder

**Comment by Biangkerok32** [comment]
@27vinayvyas can you add/make save settings after choosing dyslexic font ?
so as not to return default font .

**Comment by 27vinayvyas** [comment]
@Biangkerok32 
Hey i didn't understand what you want to convey. Save button for saving what.
Do you mean to have dyslexic font when you open the App Inventor next time after closing it??

**Comment by kangris-sket** [comment]
> @Biangkerok32
> Do you mean to have dyslexic font when you open the App Inventor next time after closing it??

Yes true .

**Comment by Biangkerok32** [comment]
> @Biangkerok32 
> Do you mean to have dyslexic font when you open the App Inventor next time after closing it??
Yes true @27vinayvyas

**Comment by ewpatton** [comment]
@27vinayvyas The end goal here would be to add a new setting to the UserSettings object that would store the flag and then restore it when the settings are loaded from the server.

**Comment by 27vinayvyas** [comment]
Hello @ewpatton  , @Biangkerok32 
I am having my end semester exams right now. I would be busy with that for another week. I will definitely try to make the required changes when i am done with them next week.

**Comment by 27vinayvyas** [comment]
Hello @ewpatton 
Can you please elaborate on exactly at which location i have to make changes
I have currently made changes in the css of the page on clicking on the button. How i am going to store these changes so that id dont get to the default on reload
Thanks

**Comment by ewpatton** [comment]
This PR was superseded by #1826

**Pull request closed without merging**